Source/DafnyLanguageServer/Workspace/NotificationPublisher.cs

```diff
@@ -51,12 +64,19 @@ public class NotificationPublisher : INotificationPublisher {
       languageServer.TextDocument.SendNotification(DafnyRequestNames.VerificationSymbolStatus, notification);
     }
 
-    private static FileVerificationStatus? GetFileVerificationStatus(DafnyDocument document) {
-      if (document.ImplementationIdToView == null || document.VerificationTasks == null) {
+    private static FileVerificationStatus? GetFileVerificationStatus(CompilationView document) {
+      if (!document.ImplementationsWereUpdated) {
+        /*
+         CompilationAfterResolution.Snapshot() gets migrated ImplementationViews.
+         It has to get migrated Diagnostics inside ImplementationViews, otherwise we get incorrect diagnostics.
+         However, migrating the ImplementationId's may mean we lose verifiable symbols, which we don't want at this point. TODO: why not?
+         To prevent publishing file verification status unless the current document has been translated,
+         the field ImplementationsWereUpdated was added.
+         */
         return null;
       }
-      return new FileVerificationStatus(document.Uri, document.Version,
-        GetNamedVerifiableStatuses(document.ImplementationIdToView));
+      return new FileVerificationStatus(document.TextDocumentItem.Uri, document.TextDocumentItem.Version,
+        GetNamedVerifiableStatuses(document.ImplementationViews));
```

Comment:
You didn't like the name ImplementationIdToViews? :-) I thought it was more explicit than a set-based name.